### ISSUE TITLE
[Bug][iOS] CollectionView doesn't fill page

### ISSUE THREAD

**Issue by rickclephas** [xamarin/Xamarin.Forms#14239]
### Description

A `CollectionView` inside a `StackLayout` inside a `TabbedPage` that is set as the `MainPage` after the app is started doesn't fill the whole page.

### Steps to Reproduce

1. Run the sample on an iPhone 
2. Click the "Go to tabbed page!" button

### Expected Behavior

The `CollectionView` fills the page.

### Actual Behavior

The `CollectionView` doesn't fill the page.

### Basic Information

- Version with issue: 5.0.0.2012
- Last known good version: n/a
- Affected Devices: iPhone 7 and iPhone 12 Pro Max (and presumably all other iPhones).

~~**Note:** this only affects physical devices, simulators aren't affected.~~

### Screenshots

![IMG_0338](https://user-images.githubusercontent.com/7353419/116987930-fa270100-accf-11eb-8d94-8abc4eeb3908.PNG)

### Reproduction Link

[XFiOSBlankSpaceIssue.zip](https://github.com/xamarin/Xamarin.Forms/files/6420276/XFiOSBlankSpaceIssue.zip)

### Workarounds

The sample contains TODOs mentioning the different parts of the code that cause this issue.
Changing either of the following will "fix" the issue.

#### Setting the MainPage to the TabbedPage

```c#
//TODO: Set MainPage to MyTabbedPage and the issue is gone
//MainPage = new MyTabbedPage();
MainPage = new MainPage();
```

#### Removing the enclosing StackLayout

```xaml
<!--TODO: Remove the enclosing StackLayout and the issue is gone-->
<StackLayout>
...
</StackLayout>
```

#### Setting the VerticalOptions to FillAndExpand

```xaml
<!--TODO: Set VerticalOptions to FillAndExpand and the issue is gone-->
<!--<CollectionView.VerticalOptions>FillAndExpand</CollectionView.VerticalOptions>-->
```

**Comment by jsuarezruiz** [comment]
Using the TabbedPage as MainPage and wrapping the CollectionView with the StackLayout, I see:
![issue14239](https://user-images.githubusercontent.com/6755973/117279804-6174ba80-ae62-11eb-9163-bb0265bab167.gif)

Seems to be correct, I am missing something?

**Comment by rickclephas** [comment]
@jsuarezruiz I wasn't able to reproduce this on a simulator either. Though I was able to reproduce this on a real iPhone 7 and 12 Pro Max. 
Didn't test on any other devices but I assume all iPhones (and maybe iPads?) are affected.

**Comment by rickclephas** [comment]
@chrome050 I am not sure if this is exactly the same issue (although definitely related!).
In our case there are 3 parts of the code that combined caused the issue. Removing/changing one of these "fixes" the issue.
I also wasn't able to reproduce this on a simulator but only on real iPhones.

In our case we fixed it with the `VerticalOptions="FillAndExpand"`, the `HeightRequest` wasn't needed.

Anyway there is definitely something very wrong with how these elements behave...

**Comment by memis1970** [comment]
Seems to be when the phone is laid flat on a table for me as mentioned in [#6908](https://github.com/xamarin/Xamarin.Forms/issues/6908)

**Comment by jfversluis** [comment]
Might be a longshot here, but would you be able to test the NuGet package from #15067 [as described here](https://github.com/xamarin/Xamarin.Forms/wiki/Preview-Packages) and let us know if this fixes this issue? 

Note that you need the specific version from that PR which in this case is 5.0.0.7996.

Thanks!

**Comment by rickclephas** [comment]
@jfversluis unfortunately 5.0.0.7996 has the same issue.

Previously I wasn't able to reproduce this with a simulator but only on physical devices. 
However even on physical devices this bug isn't always occurring. 
It turns out that this issue is caused by the **orientation** of the device (like @memis1970 mentioned).

This is also the difference between a physical device and the simulator that by default prevents the issue from occurring on a simulator. Changing the orientation of the simulator to "Face Up" makes this issue reproducible on a simulator as well.

See the following video that demonstrates the issue with:
- The reproduction code in XFiOSBlankSpaceIssue.zip
- Xamarin Forms 5.0.0.7996
- iOS 15.2 iPhone 12 Pro Max Simulator

https://user-images.githubusercontent.com/7353419/150502554-94865677-3d92-443c-aaad-221f50edab4a.mp4

**Comment by jfversluis** [comment]
Thank you for checking and the additional info!

**Comment by rickclephas** [comment]
I am seeing the same result with the 5.0.0.2337 NuGet package from that release.

**Comment by rickclephas** [comment]
@jfversluis @jsuarezruiz it seems that when the orientation is "Face Up" the measurement logic interprets that as "Landscape". If you look at the screenshot and video above you'll notice that the blue area is actually a perfect square.

**Comment by jfversluis** [comment]
Yeah that definitely seems like a good point. Not sure if we actually do that much ourselves with the orientation though, but this is a good lead.

**Comment by brendonupson** [comment]
I'm having similar issue with an app. In my testing, launching the app when the phone is flat on a table produces the blank area. Kill the app and relaunch it in vertical / portrait mode, no blank area. Explains why always seemed ok in the sim. @rickclephas seems to have nailed the issue. This is in v5.0.0.2337

**Comment by rickclephas** [comment]
Can confirm that as mentioned in https://github.com/xamarin/Xamarin.Forms/issues/6908#issuecomment-1094125046 this has been fixed in 5.0.0.2401 by https://github.com/xamarin/Xamarin.Forms/pull/13163.

**Comment by jfversluis** [comment]
Thanks for the confirmation @rickclephas !
